server/src/main/java/org/candlepin/async/JobArguments.java

```diff
@@ -14,176 +14,185 @@
  */
 package org.candlepin.async;
 
-import java.util.Collection;
-import java.util.Iterator;
+import org.candlepin.util.ObjectMapperFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
 import java.util.Map;
 import java.util.Set;
 
 
 
 /**
- * The JobDataMap is a typed HashMap that provides utility methods for fetching values as various
- * datatypes.
+ * The JobArguments is a map-like view of the arguments provided to the job during construction.
+ * Unlike a typical map, the types of the values must be known to properly fetch them.
  */
-public class JobDataMap implements Map<String, Object> {
+public class JobArguments {
+    private static ObjectMapper mapper = ObjectMapperFactory.getObjectMapper();
 
-    final Map<String, Object> map;
+    private Map<String, String> data;
 
     /**
-     * Creates a new JobDataMap instance backed by the provided map.
+     * Creates a new serialized map using the specified map as a data store.
      *
      * @param map
-     *  The map to use as the backing map
-     *
-     * @throws IllegalArgumentException
-     *  if the provided map is null
+     *  the map to use as the data store for this map
      */
-    public JobDataMap(Map<String, Object> map) {
+    public JobArguments(Map<String, String> map) {
         if (map == null) {
             throw new IllegalArgumentException("map is null");
         }
 
-        this.map = map;
+        this.data = map;
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void clear() {
-        this.map.clear();
-    }
+    // /**
+    //  * Serializes the given object into a JSON string.
+    //  */
+    // private String serialize(Object value) {
+    //     try {
+    //         return value != null ? mapper.writeValueAsString(value) : null;
+    //     }
+    //     catch (JsonProcessingException e) {
+    //         throw new RuntimeException("Unable to serialize object: " + value, e);
+    //     }
+    // }
```

Comment:
I think I'm going to restore this method and add a helper method to JobArguments:

```
public static JobArguments fromMap(Map<String, Object> map) {
  ...
}
```

Maybe also move the creation of the arguments to AsyncJobStatus so it's entirely encapsulated there and there isn't any extra juggling of the data types for internal storage.

Reply:
I've restored this method and made it public to be usable by the other methods. Not sure if this is the best way to handle it, but I wanted to avoid having different ObjectMapper instances serializing and deserializing.